specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json

```diff
@@ -0,0 +1,453 @@
+              "$ref": "#/definitions/InstancePoolUpdate"
+            }
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the instance pool.",
+            "schema": {
+              "$ref": "#/definitions/InstancePool"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidResourceRequestBody - The resource or resource properties in the request body is empty or invalid."
+          },
+          "202": {
+            "description": "Accepted"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Patch an instance pool": {
+            "$ref": "./examples/PatchInstancePool.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools": {
+      "get": {
+        "tags": [
+          "InstancePools"
+        ],
+        "description": "Gets a list of instance pools in the resource group",
+        "operationId": "InstancePools_ListByResourceGroup",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the list of instance pools.",
+            "schema": {
+              "$ref": "#/definitions/InstancePoolListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List instance pools by resource group": {
+            "$ref": "./examples/ListInstancePools.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InstancePoolProperties": {
+      "description": "Properties of an instance pool.",
+      "required": [
+        "subnetId",
+        "vCores",
+        "licenseType"
+      ],
+      "type": "object",
+      "properties": {
+        "subnetId": {
+          "description": "Resource ID of the subnet to place this instance pool in.",
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "vCores": {
+          "format": "int32",
+          "description": "Count of vCores belonging to this instance pool.",
+          "type": "integer",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "licenseType": {
```

Comment:
Can this be an enum? Is the list of allowed values finite?

Reply:
The list of values are finite for now [LicenseIncluded / BasePrice]
We are following managedInstances design.

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json#L341

Reply:
Enum could be nice, but making this consistent between managedInstances and instancePools would be desired.
Let me know if you think making it inconsistent is okay.

Reply:
@tomapaunovic is making changes in this area for MI. Please sync with him. They really should both be using enum (model as string = true).

Reply:
Oh awesome, thanks for letting us know @jaredmoo.
I'll sync with @tomapaunovic about this. Can probably re-use his license types once he has checked in.

Reply:
Made updates in latest commit.